### ISSUE TITLE
dismiss(DialogFragment) after pause/resume crashes

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/android#279]
### Steps to reproduce
1. Enable the app passcode
2. Open the share view and share a file/folder by link
3. Minimize the app using the device main button with the share view in foreground
4. Open again to the app

### Actual behaviour
Passcode view is displayed and app crashes

### Expected behaviour
Passcode view is displayed

### Stack Trace
```
8-10 11:32:06.187 25328-25328/com.owncloud.android E/AndroidRuntime: FATAL EXCEPTION: main
  Process: com.owncloud.android, PID: 25328
  java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
      at android.support.v4.app.FragmentManagerImpl.checkStateLoss(FragmentManager.java:1489)
      at android.support.v4.app.FragmentManagerImpl.enqueueAction(FragmentManager.java:1507)
      at android.support.v4.app.BackStackRecord.commitInternal(BackStackRecord.java:634)
      at android.support.v4.app.BackStackRecord.commit(BackStackRecord.java:613)
      at android.support.v4.app.DialogFragment.dismissInternal(DialogFragment.java:201)
      at android.support.v4.app.DialogFragment.dismiss(DialogFragment.java:167)
      at com.owncloud.android.ui.activity.FileActivity.dismissLoadingDialog(FileActivity.java:934)
      at com.owncloud.android.ui.activity.FileActivity.onRemoteOperationFinish(FileActivity.java:753)
      at com.owncloud.android.ui.activity.ShareActivity.onRemoteOperationFinish(ShareActivity.java:220)
      at com.owncloud.android.utils.GetShareWithUsersAsyncTask.onPostExecute(GetShareWithUsersAsyncTask.java:94)
      at com.owncloud.android.utils.GetShareWithUsersAsyncTask.onPostExecute(GetShareWithUsersAsyncTask.java:44)
      at android.os.AsyncTask.finish(AsyncTask.java:651)
      at android.os.AsyncTask.access$500(AsyncTask.java:180)
      at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:668)
      at android.os.Handler.dispatchMessage(Handler.java:102)
      at android.os.Looper.loop(Looper.java:158)
      at android.app.ActivityThread.main(ActivityThread.java:7225)
      at java.lang.reflect.Method.invoke(Native Method)
      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)

```

**Comment by tobiasKaminsky** [comment]
What is the PR?

**Comment by AndyScherzinger** [comment]
it is in #281 - can also be shipped with 1.3.1 if we want to (I vote yes). I'll leave the decision with you :)
